kymatio/scattering1d/core/scattering1d.py

```diff
@@ -0,0 +1,159 @@
+
+
+def scattering1d(x, pad, unpad, backend, J, psi1, psi2, phi, pad_left=0, pad_right=0,
+               ind_start=None, ind_end=None, oversampling=0,
+               max_order=2, average=True, size_scattering=(0, 0, 0), vectorize=False):
+    """
+    Main function implementing the 1-D scattering transform.
+
+    Parameters
+    ----------
+    x : Tensor
+        a torch Tensor of size `(B, 1, T)` where `T` is the temporal size
+    psi1 : dictionary
+        a dictionary of filters (in the Fourier domain), with keys (`j`, `q`).
+        `j` corresponds to the downsampling factor for
+        :math:`x \\ast psi1[(j, q)]``, and `q` corresponds to a pitch class
+        (chroma).
+        * psi1[(j, n)] is itself a dictionary, with keys corresponding to the
+        dilation factors: psi1[(j, n)][j2] corresponds to a support of size
+        :math:`2^{J_\\text{max} - j_2}`, where :math:`J_\\text{max}` has been
+        defined a priori (`J_max = size` of the padding support of the input)
+        * psi1[(j, n)] only has real values;
+        the tensors are complex so that broadcasting applies
+    psi2 : dictionary
+        a dictionary of filters, with keys (j2, n2). Same remarks as for psi1
+    phi : dictionary
+        a dictionary of filters of scale :math:`2^J` with keys (`j`)
+        where :math:`2^j` is the downsampling factor.
+        The array `phi[j]` is a real-valued filter.
+    J : int
+        scale of the scattering
+    pad_left : int, optional
+        how much to pad the signal on the left. Defaults to `0`
+    pad_right : int, optional
+        how much to pad the signal on the right. Defaults to `0`
+    ind_start : dictionary of ints, optional
+        indices to truncate the signal to recover only the
+        parts which correspond to the actual signal after padding and
+        downsampling. Defaults to None
+    ind_end : dictionary of ints, optional
+        See description of ind_start
+    oversampling : int, optional
+        how much to oversample the scattering (with respect to :math:`2^J`):
+        the higher, the larger the resulting scattering
+        tensor along time. Defaults to `0`
+    order2 : boolean, optional
+        Whether to compute the 2nd order or not. Defaults to `False`.
+    average_U1 : boolean, optional
+        whether to average the first order vector. Defaults to `True`
+    size_scattering : tuple
+        Contains the number of channels of the scattering, precomputed for
+        speed-up. Defaults to `(0, 0, 0)`.
+    vectorize : boolean, optional
+        whether to return a dictionary or a tensor. Defaults to False.
+
+    """
+    subsample_fourier = backend.subsample_fourier
+    modulus_complex = backend.modulus_complex
+    fft1d_c2c = backend.fft1d_c2c
+    ifft1d_c2c = backend.ifft1d_c2c
+    real = backend.real
+    finalize = backend.finalize
+
+
+    # S is simply a dictionary if we do not perform the averaging...
+    if vectorize:
```

Comment:
This choice between dictionary stuff and vectorize stuff makes the code messy, and makes the structure of this 1d scattering transform wildly different from the 2d scattering transform (which seems to assume the vectorization). 

Should we remove the dictionary stuff we do here, or do we wanna add it to the rest of the scattering transforms, or keep as it?

Reply:
@janden ? I'm ok to keep it as it is for now: we should aim to unifying the code in a second step.